internal/exp/metrics/streams/streams.go

```diff
@@ -56,3 +61,17 @@ func (m HashMap[T]) Len() int {
 type Evictor interface {
 	Evict() identity.Stream
 }
+
+type DataPointSlice[DP DataPoint[DP]] interface {
+	Len() int
+	At(i int) DP
+	AppendEmpty() DP
+}
+
+type DataPoint[Self any] interface {
+	pmetric.NumberDataPoint | pmetric.HistogramDataPoint | pmetric.ExponentialHistogramDataPoint | pmetric.SummaryDataPoint
```

Comment:
I think this is overly thight. any type satisfying below characteristics should work (e.g. a custom wrapper of above)